nfc/NfcCx/Cx/NfcCxESE.cpp

```diff
@@ -0,0 +1,1888 @@
+/*++
+
+Copyright (c) Microsoft Corporation.  All Rights Reserved
+
+Module Name:
+
+    NfcCxESE.cpp
+
+Abstract:
+
+    Embedded SE Interface definition
+
+Environment:
+
+    User-mode Driver Framework
+
+--*/
+
+#include "NfcCxPch.h"
+
+#include "NfcCxESE.tmh"
+
+// TODO: Temporarily define this here until we get ScDeviceEnum.h published
+// {D6B5B883-18BD-4B4D-B2EC-9E38AFFEDA82}, 2, DEVPROP_TYPE_BYTE
+DEFINE_DEVPROPKEY(DEVPKEY_Device_ReaderKind,
+    0xD6B5B883, 0x18BD, 0x4B4D, 0xB2, 0xEC, 0x9E, 0x38, 0xAF, 0xFE, 0xDA, 0x82, 0x02);
+
+#define SCARD_READER_TYPE_EMBEDDEDSE 0x800
+
+typedef
+NTSTATUS
+NFCCX_EmbeddedSE_DISPATCH_HANDLER(
+    _In_ WDFDEVICE Device,
+    _In_ WDFREQUEST Request,
+    _In_opt_bytecount_(InputBufferLength) PVOID InputBuffer,
+    _In_ size_t InputBufferLength,
+    _Out_opt_bytecap_(OutputBufferLength) PVOID OutputBuffer,
+    _In_ size_t OutputBufferLength
+    );
+
+typedef NFCCX_EmbeddedSE_DISPATCH_HANDLER *PFN_NFCCX_EmbeddedSE_DISPATCH_HANDLER;
+
+typedef struct _NFCCX_eSE_DISPATCH_ENTRY {
+    ULONG IoControlCode;
+    BOOLEAN fPowerManaged;
+    BOOLEAN fSequentialDispatch;
+    size_t MinimumInputBufferLength;
+    size_t MinimumOutputBufferLength;
+    PFN_NFCCX_EmbeddedSE_DISPATCH_HANDLER DispatchHandler;
+} NFCCX_eSE_DISPATCH_ENTRY, *PNFCCX_eSE_DISPATCH_ENTRY;
+
+NFCCX_SC_DISPATCH_HANDLER NfcCxEmbeddedSEInterfaceDispatchTransmit;
+
+typedef struct _EmbeddedSE_ATTRIBUTE_DISPATCH_ENTRY {
+    DWORD dwAttributeId;
+    PBYTE pbResultBuffer;
+    size_t cbResultBuffer;
+    BOOL fRequireLock;
+    PFN_NFCCX_SC_ATTRIBUTE_DISPATCH_HANDLER pfnDispatchHandler;
+} EmbeddedSE_ATTRIBUTE_DISPATCH_ENTRY, *PEmbeddedSEATTRIBUTE_DISPATCH_ENTRY;
+
+static const CHAR eSEReaderVendorName[] = "NXP";
```

Comment:
I believe this should be part of the Client, just like other controller-proprietary values/logic.
Code in this file should be general and more vendor-agnostic